src/vm.rs

```diff
@@ -279,15 +282,21 @@ impl VM {
     }
 
     fn set_pc(&mut self, new_pc: usize) {
+        let new_pc = u16::try_from(new_pc).expect("PC should fit in 16 bits");
         self.registers[REG_IDX_PC] = new_pc as u16;
     }
 
     fn advance_pc(&mut self) {
-        self.registers[REG_IDX_PC] += 1;
+        self.set_pc(self.pc_with_offset(1));
     }
 
     fn pc_with_offset(&self, offset: i16) -> usize {
-        (self.pc() as i16).wrapping_add(offset) as usize // TODO is this ok?
+        let pc = self.pc() as u16;
+        if offset >= 0 {
+            pc.wrapping_add(offset as u16) as usize
+        } else {
+            pc.wrapping_sub((-offset) as u16) as usize
+        }
```

Comment:
This looks weird assuming offset is negative `pc.wrapping_add(negative_num)` is the same as `pc.wrapping_sub(-negative_num)` adding a negative number should be the same as substracting a positive number

Reply:
Done, I simplified it.